### PR TITLE
Add refresh back-off period configuration

### LINKED ISSUE #1912: [on-prem] Update is skipped completely if there was previous issue
Hi there
We're running scala-steward on prem (github) and noticed that once it stopped sending updates per particular repos:

```
2021-02-01 10:11:22,810 INFO  ──────────── Steward {org}/{proj-name} ────────────
2021-02-01 10:11:22,818 INFO  Check cache of {org}/{proj-name}
2021-02-01 10:11:22,842 DEBUG Write /opt/ag-scala-steward-workplace/workspace/store/refresh_error/v1/github/{org}/{proj-name}/refresh_error.json
2021-02-01 10:11:22,846 ERROR Steward {org}/{proj-name} failed
org.scalasteward.core.repocache.RepoCacheAlg$$anon$1: Skipping due to previous error
```

checking error log in `/opt/ag-scala-steward-workplace/workspace/store/refresh_error/v1/github/{org}/{proj-name}/refresh_error.json`:
```
{
  "failedAt" : 1611962373257,
  "message" : "'SBT_OPTS=-Xmx4096m -Xss8m -XX:MaxMetaspaceSize=1024m sbt -Dsbt.color=false -Dsbt.log.noformat=true -Dsbt.supershell=false ;+ stewardDependencies;reload plugins;stewardDependencies' exited with code 1\n[info] welcome to sbt 1.3.13 (Oracle Corporation Java 1.8.0_252)\n[info] loading global plugins from /home/ag-steward-svc/.sbt/1.0/plugins\n[info] loading settings for project {proj-name} from plugins.sbt ...\n[info] loading project definition from /opt/ag-scala-steward-workplace/workspace/repos/{org}/{proj-name} project\n[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.\n[info] loading settings for project root from build.sbt,version.sbt ...\n[info] loading settings for project cucumber from build.sbt ...\n[info] set current project to {proj-name} (in build file:/opt/ag-scala-steward-workplace/workspace/repos/{org}/{proj-name}/)\n[error] Expected ';'\n[error] Not a valid command: stewardDependencies\n[error] Not a valid project ID: stewardDependencies\n[error] Expected ':'\n[error] Not a valid key: stewardDependencies (similar: allDependencies, rpmDependencies, excludeDependencies)\n[error] + stewardDependencies\n[error]                      ^\n"
```

Removing `/opt/ag-scala-steward-workplace/workspace/store/refresh_error/v1/github/{org}/{proj-name}/refresh_error.json` seems solved the problem and subsequent update is successful

Is it expected behavior?

### PR DESCRIPTION
**Pull request by javierarrieta**
Add refresh back-off period configuration

Fixes #1912

### CODE REVIEW

File: modules/core/src/main/scala/org/scalasteward/core/application/Cli.scala [hunk 1]
```diff
@@ -58,7 +58,8 @@ object Cli {
       githubAppKeyFile: Option[File] = None,
       githubAppId: Option[Long] = None,
       urlCheckerTestUrl: Option[Uri] = None,
-      defaultMavenRepo: Option[String] = None
+      defaultMavenRepo: Option[String] = None,
+      refreshBackoffPeriod: Option[FiniteDuration] = None
```
Comment (javierarrieta) [inline modules/core/src/main/scala/org/scalasteward/core/application/Cli.scala:62]: I wasn't sure if set the default at this level or when creating the `Config`. Might make more sense to set it here than where I wrote it

**Comment by javierarrieta** [comment]
I will add the config to the docs when we agree on the approach

File: modules/core/src/main/scala/org/scalasteward/core/application/Cli.scala (hunk 1 above)
Comment (fthomas) [reply modules/core/src/main/scala/org/scalasteward/core/application/Cli.scala:62]: Most default values are set here, so I guess it makes sense to also set it here for `refreshBackoffPeriod`.

**Review by fthomas — APPROVED** [review]
This LGTM, many thanks @javierarrieta.

So to disable any refresh backoff, one just passes `--refresh-backoff-period 0days` as CLI option.

File: modules/core/src/main/scala/org/scalasteward/core/application/Cli.scala (hunk 1 above)
Comment (javierarrieta) [reply modules/core/src/main/scala/org/scalasteward/core/application/Cli.scala:62]: Yeah, I will change that, thanks!

**Comment by javierarrieta** [comment]
> This LGTM, many thanks @javierarrieta.
> 
> So to disable any refresh backoff, one just passes `--refresh-backoff-period 0days` as CLI option.

Yes, I will add it to the docs in this PR, let's see if I fixed the build

**Comment by javierarrieta** [comment]
Documentation added to the help.md file, would that be enough?

**Comment by fthomas** [comment]
> Documentation added to the help.md file, would that be enough?

Yes, thanks!

**Comment by javierarrieta** [comment]
Thanks for the quick turnaround and mostly thanks for this great piece of software that help many of us with their day to day

**Pull request merged**